feature-find-meal/src/main/java/com/kinandcarta/feature/find/meal/view/FindMealFragment.kt

```diff
@@ -1,115 +1,102 @@
 package com.kinandcarta.feature.find.meal.view
 
-import android.annotation.SuppressLint
-import android.content.Context
+import android.app.Dialog
 import android.os.Bundle
+import android.util.Log
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import android.widget.FrameLayout
+import androidx.coordinatorlayout.widget.CoordinatorLayout
+import androidx.core.view.get
+import androidx.core.view.isEmpty
 import androidx.core.view.isVisible
 import androidx.fragment.app.Fragment
 import androidx.fragment.app.viewModels
+import androidx.recyclerview.widget.LinearLayoutManager
 import androidx.recyclerview.widget.RecyclerView
-import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment
 import com.kinandcarta.feature.find.meal.R
 import com.kinandcarta.feature.find.meal.adapter.MealsAdapter
-import com.kinandcarta.feature.find.meal.extension.requestFineLocationPermission
 import com.kinandcarta.feature.find.meal.extension.showToast
-import com.kinandcarta.feature.find.meal.utility.PermissionResultParser
-import com.kinandcarta.feature.find.meal.viewmodel.FindMealViewModel
+import com.kinandcarta.feature.find.meal.viewmodel.DisplayMealsViewModel
 import dagger.hilt.android.AndroidEntryPoint
 
 @AndroidEntryPoint
 class FindMealFragment : Fragment() {
 
     private val progressBarView: FrameLayout by lazy { requireView().findViewById(R.id.progressBar) }
-    private val viewModel: FindMealViewModel by viewModels()
-
-    private val mealsAdapter = MealsAdapter() { id, position -> viewModel.reserveAMeal(id, position)}
-
-    override fun onAttach(context: Context) {
-        super.onAttach(context)
-        requestFineLocationPermission(LOCATION_PERMISSION_REQUEST_CODE)
-    }
+    private val viewModel: DisplayMealsViewModel by viewModels()
+    private lateinit var mealsRecyclerView: RecyclerView
```

Comment:
```suggestion
    private val mealsRecyclerView: RecyclerView by lazy { view.findViewById(R.id.rvMeals) }
```

This way, your mealsRecyclerView is immutable